### PR TITLE
Refactor usages of global `Homebrew.args`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Pass `args` down the call chain instead of using global state everywhere. This should make refactoring `CLI::Parser` possible in order to use it for `cask` subcommands.

### CODE REVIEW

File: Library/Homebrew/extend/ENV/shared.rb [hunk 1]
```diff
@@ -253,9 +253,9 @@ def fortran
   end
 
   # @private
-  def effective_arch
-    if Homebrew.args.build_bottle? && Homebrew.args.bottle_arch
-      Homebrew.args.bottle_arch.to_sym
+  def effective_arch(args)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/extend/ENV/shared.rb:256]: When is this called with `args` unset?

File: Library/Homebrew/fetch.rb [hunk 2]
```diff
@@ -1,13 +1,13 @@
 # frozen_string_literal: true
 
 module Homebrew
-  module Fetch
+  module FetchHelpers
```
Comment (MikeMcQuaid) [inline Library/Homebrew/fetch.rb:4]: Why the rename? Feels a bit inconsistent with other similar modules.

File: Library/Homebrew/extend/ENV/shared.rb (hunk 1 above)
Comment (reitermarkus) [reply Library/Homebrew/extend/ENV/shared.rb:256]: There are some places in `Requirement`s, I think one place is in `java_dep_if_needed`, where there is no way to easily pass down the `args`. And in those cases I don't think the value of `effective_arch` is used.

**Comment by reitermarkus** [comment]
> Is it worth removing the global `Homebrew.args` entirely to avoid usage?

The goal is to have multiple levels of args:

1. `brew` level: e.g. `--debug`, `--verbose` (which is now the global `Homebrew.args`)
2. `brew <subcommand>` level
3. `brew cask <subcommand>` level

File: Library/Homebrew/extend/ENV/shared.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/extend/ENV/shared.rb:256]: Cool, fine with me assuming it works as expected 👍

**Comment by MikeMcQuaid** [comment]
> * `brew` level: e.g. `--debug`, `--verbose` (which is now the global `Homebrew.args`)
> * `brew <subcommand>` level

This sounds good. Still think I it may be worth having a slightly different API at this time to avoid `Homebrew.args` being called in new code and getting unexpected results.

> 3\. `brew cask <subcommand>` level

Why would this need to be separate from the `brew` level, out of interest?

**Comment by reitermarkus** [comment]
> Why would this need to be separate from the `brew` level, out of interest?

There are some flags which all `cask` commands support, e.g. `--appdir`, and there are  more specific flags for `cask` subcommands. Maybe this can also be refactored then.

**Pull request merged**